src/telem/lib/input/mekanism/DigitalMinerInputAdapter.lua

```diff
@@ -0,0 +1,67 @@
+local o = require 'telem.lib.ObjectModel'
+local t = require 'telem.lib.util'
+
+local InputAdapter      = require 'telem.lib.InputAdapter'
+local Metric            = require 'telem.lib.Metric'
+local MetricCollection  = require 'telem.lib.MetricCollection'
+
+local DigitalMinerInputAdapter = o.class(InputAdapter)
+DigitalMinerInputAdapter.type = 'DigitalMinerInputAdapter'
+
+function DigitalMinerInputAdapter:constructor (peripheralName, categories)
+    self:super('constructor')
+
+    -- TODO this will be a configurable feature later
+    self.prefix = 'mekdigitalminer:'
+
+    -- TODO make these constants
+    local allCategories = {
+        'basic',
+    }
+
+    if not categories then
+        self.categories = { 'basic' }
+    elseif categories == '*' then
+        self.categories = allCategories
+    else
+        self.categories = categories
+    end
+
+    -- boot components
+    self:setBoot(function ()
+        self.components = {}
+
+        self:addComponentByPeripheralID(peripheralName)
+    end)()
+end
+
+function DigitalMinerInputAdapter:read ()
+    self:boot()
+
+    local source, miner = next(self.components)
+
+    local metrics = MetricCollection()
+
+    local loaded = {}
+
+    for _,v in ipairs(self.categories) do
+        -- skip, already loaded
+        if loaded[v] then
+            -- do nothing
+
+        -- Literally all we have lmao
+        elseif v == 'basic' then
+            metrics:insert(Metric{ name = self.prefix .. 'energy', value = mekanismEnergyHelper.joulesToFE(miner.getEnergy()), unit = "FE", source = source })
+            metrics:insert(Metric{ name = self.prefix .. 'energy_filled_percentage', value = (miner.getEnergyFilledPercentage()), unit = nil, source = source })
+            metrics:insert(Metric{ name = self.prefix .. 'energy_usage', value = mekanismEnergyHelper.joulesToFE(miner.getEnergyUsage()), unit = "FE/t", source = source })
+            metrics:insert(Metric{ name = self.prefix .. 'get_to_mine', value = miner.getToMine(), unit = "items", source = source })
+            metrics:insert(Metric{ name = self.prefix .. 'is_running', value = (miner.isRunning() and 1 or 0), unit = nil, source = source })
```

Comment:
Remove function verb from name, recommend `running`.

Thought about this one for a bit, as I do like labelling true/false flags differently. But, Induction Matrix adapter does this for `isFormed() => formed`, so probably better to follow the precedent.